### PR TITLE
[MetaSchedule] Filter vector_load_lens based on buffer dtype

### PR DESCRIPTION
**Pull request by vinx13**
This makes the same config generic to work across workloads with different types.

cc @junrushao1994 @zxybazh @spectrometerHBH @masahi

### CODE REVIEW

File: tests/python/unittest/test_meta_schedule_schedule_rule_multi_level_tiling.py [hunk 1]
```diff
@@ -883,13 +883,13 @@ def test_cuda_tensor_core_matmul_relu_global():
...
 l77 = sch.fuse(l75, l76, preserve_unit_iters=True)
-v78 = sch.sample_categorical(candidates=[1, 2, 3, 4], probs=[0.25, 0.25, 0.25, 0.25])
+v78 = sch.sample_categorical(candidates=[1, 2, 4, 8], probs=[0.25, 0.25, 0.25, 0.25])
 sch.annotate(block_or_loop=b70, ann_key="meta_schedule.cooperative_fetch", ann_val=v78)
 b79 = sch.cache_read(block=b19, read_buffer_index=1, storage_scope="shared")
 sch.compute_at(block=b79, loop=l46, preserve_unit_loops=True)
 l80, l81, l82, l83, l84, l85 = sch.get_loops(block=b79)
 l86 = sch.fuse(l84, l85, preserve_unit_iters=True)
-v87 = sch.sample_categorical(candidates=[1, 2, 3, 4], probs=[0.25, 0.25, 0.25, 0.25])
+v87 = sch.sample_categorical(candidates=[1, 2, 4, 8], probs=[0.25, 0.25, 0.25, 0.25])
```
Comment (junrushao1994) [inline tests/python/unittest/test_meta_schedule_schedule_rule_multi_level_tiling.py:892]: BTW, i'm not sure if we should rule out `float3` in CUDA

Comment (vinx13) [reply tests/python/unittest/test_meta_schedule_schedule_rule_multi_level_tiling.py:892]: This is fp16

**Comment by vinx13** [comment]
@tvm-bot rerun

File: tests/python/unittest/test_meta_schedule_schedule_rule_multi_level_tiling.py (hunk 1 above)
Comment (junrushao1994) [reply tests/python/unittest/test_meta_schedule_schedule_rule_multi_level_tiling.py:892]: That makes sense!

**Review by junrushao1994 — APPROVED** [review]
LGTM

**Pull request merged**